sql/core.go

```diff
@@ -178,6 +178,11 @@ type PrimaryKeySchemaTarget interface {
 	WithPrimaryKeySchema(schema PrimaryKeySchema) (Node, error)
 }
 
+type DynamicColumnsTable interface {
```

Comment:
Needs a comment, same with method names

Reply:
Should also give this a boolean `HasDynamicColumns() bool` to make it clearer as a marker interface